cmd/cmd/userAdd.go

```diff
@@ -0,0 +1,102 @@
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+ */
+
+package cmd
+
+import (
+    "fmt"
+    "github.com/spf13/cobra"
+    "github.com/wso2/product-mi-tooling/cmd/utils"
+)
+
+// Show user command related usage info
+const addUserCmdLiteral = "add"
+const addUserCmdShortDesc = "Add new user"
+const addUserCmdLongDesc = "Add new user to the micro-integrator user store\n"
+
+var addUserCmdExamples = "Example:\n" +
+    "To Add a new user\n" +
+    "  " + programName + " " + usersCmdLiteral + " " + addUserCmdLiteral + " [new user-id] [password] [is-admin]\n" +
+    "  " + programName + " " + usersCmdLiteral + " " + addUserCmdLiteral + " user1 passw0rd true\n"
+
+// userAddCmd represents the add users command
+var userAddCmd = &cobra.Command{
+    Use:   addUserCmdLiteral,
+    Short: addUserCmdShortDesc,
+    Long:  addUserCmdLongDesc + addUserCmdExamples,
+    Run: func(cmd *cobra.Command, args []string) {
+        handleAddUserCmdArguments(args)
+    },
+}
+
+func init() {
+    userAddCmd.SetHelpTemplate(addUserCmdLongDesc + "Usage:\n" +
+        "  " + programName + " " + usersCmdLiteral + " " + addUserCmdLiteral +
+        " [new user-id] [password] [is-admin]\n" +
+        addUserCmdExamples + utils.GetCmdFlags(addUserCmdLiteral))
+    usersCmd.AddCommand(userAddCmd)
+}
+
+func handleAddUserCmdArguments(args []string) {
+    utils.Logln(utils.LogPrefixInfo + "Add user called")
+    if len(args) == 1 {
+        if args[0] == "help" {
+            printAddUserHelp()
+        } else {
+            fmt.Println("Invalid arguments. See the usage below")
+            printAddUserHelp()
+        }
+    } else if len(args) == 3 {
+        var userId = args[0]
+        var password = args[1]
+        var isAdmin = args[2]
+        if !(isAdmin == "false" || isAdmin == "true") {
+            fmt.Println("[is-admin] parameter should be either true or false")
+        }
+        executeAddUserCmd(userId,password,isAdmin)
+    } else {
+        fmt.Println("Invalid number of arguments. See the usage below")
+        printAddUserHelp()
+    }
+}
+
+func printAddUserHelp() {
+    fmt.Print(addUserCmdLongDesc + "Usage:\n" +
+        "  " + programName + " " + usersCmdLiteral + " " + addUserCmdLiteral +
+        " [new user-id] [password] [is-admin]\n" +
+        addUserCmdExamples + utils.GetCmdFlags(usersCmdLiteral))
+}
+
+func executeAddUserCmd(userId string, password string, isAdmin string) {
+    headers := make(map[string]string)
+    headers["Content-Type"] = "application/json"
+
+    body := make(map[string]string)
+    body["userId"] = userId
+    body["password"] = password
+    body["isAdmin"] = isAdmin
+
+    finalUrl := utils.GetRESTAPIBase() + utils.PrefixUsers
+    res, err := utils.InvokePOSTRequest(finalUrl, headers, body)
+    fmt.Println(res)
+    if err == nil {
```

Comment:
Here if the POST request returns a 404, it is not an error. So we cannot say that we have created successfully.

Reply:
+1 fixed